cloud/gcp/modules/storage/main.tf

```diff
@@ -0,0 +1,75 @@
+resource "google_kms_key_ring" "keyring" {
+  name     = "civiform-keyring"
+  location = var.region
+}
+
+resource "google_kms_crypto_key" "storage_key" {
+  name            = "gcs-key"
+  key_ring        = google_kms_key_ring.keyring.id
+  rotation_period = "604800s" //7 days
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+// Need the gcs service account name beforehand.
+data "google_storage_project_service_account" "gcs_account" {
+}
+
+// Allow gcs service account on key.
+resource "google_kms_crypto_key_iam_binding" "kms_binding" {
+  crypto_key_id = google_kms_crypto_key.storage_key.id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  members = ["serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"]
+}
+
+
+resource "google_storage_bucket" "file_storage" {
+  name          = "cloud-filestore-${var.application_name_postfix}"
```

Comment:
nit: is filestore not a different thing than gcs?

Reply:
you're right. Changed the name